src/plot_api/plot_api.js

```diff
@@ -1607,10 +1466,6 @@ function _restyle(gd, aobj, _traces) {
             else if(ai === 'colorbar.tick0' || ai === 'colorbar.dtick') {
                 doextra('colorbar.tickmode', 'linear', i);
```

Comment:
Why does this remain? Should this get done automatically via `impliedEdits`?

Reply:
Oh good catch - yes, the impliedEdits are there, I'll :hocho: this. One more down!

Though I notice from looking at `PlotSchema.get()` that the `{attr: undefined}` entries in `impliedEdits` (needed just for undo/redo) get stripped by `extendDeep`... not sure if that's worth worrying about, I wouldn't want to mangle `extendDeep` just for that purpose.

Reply:
We do have another extend flavor [`extendDeepAll`](https://github.com/plotly/plotly.js/blob/11bb997e78128b9f47db2ed392982f82a026e589/src/lib/extend.js#L37) which I think does just that: deep extends and keeps `undefined` values.

Reply:
Nice, I didn't know about `extendDeepAll`, but that's exactly it - if you need to do a deep extend with attributes, it now needs to be changed to `extendDeepAll`. https://github.com/plotly/plotly.js/pull/1999/commits/238e2482f06810fd759d14aebee386724c0e1e45 does this, and also removes some of the deep extends entirely 🐎  - along with a test that we don't have any empty `impliedEdits`.

I'll also note that over the course of this PR I found a lot of places where we were doing
```js
extend(Flat|Deep)({}, {some object we're creating here}, {more stuff})
```
The first `{}` is unnecessary, because we don't mind mutating a new and disposable object. I removed it whenever I noticed this 🐎